tests/test_tokenization.py

```diff
@@ -9,12 +15,303 @@ def _assert_tokenization(tokenizer: TokenizerWrapper):
+                truncation=False,
+                padding=False,
+                max_length=None,
+                special_tokens={"pad_token": "[PAD]"},
+            ),
+            1030,
+            0,
+        ),
+        (
+            "AAAAAAAA" * 1030,
+            PreTrainedHFTokenizerConfig(
+                pretrained_model_name_or_path="data/tokenizer/hf_gpt2",
+                truncation=False,
+                padding=True,
+                max_length=None,
+                special_tokens={"pad_token": "[PAD]"},
+            ),
+            1030,
+            0,
+        ),
+        (
+            "AAAAAAAA" * 1030,
+            PreTrainedHFTokenizerConfig(
+                pretrained_model_name_or_path="data/tokenizer/hf_gpt2",
+                truncation=True,
+                padding=True,
+                max_length=None,
+                special_tokens={"pad_token": "[PAD]"},
+            ),
+            1024,
+            0,
+        ),
+        # if we want to pad to max model input length we have to use padding="max_length",
+        # otherwise we will only pad to the longest sequence in the batch.
+        # see: https://huggingface.co/docs/transformers/pad_truncation#padding-and-truncation
+        (
+            "AAAAAAAA" * 1020,
+            PreTrainedHFTokenizerConfig(
+                pretrained_model_name_or_path="data/tokenizer/hf_gpt2",
+                truncation=True,
+                padding=True,
+                max_length=None,
+                special_tokens={"pad_token": "[PAD]"},
+            ),
+            1020,
+            0,
+        ),
+        (
+            "AAAAAAAA" * 1020,
+            PreTrainedHFTokenizerConfig(
+                pretrained_model_name_or_path="data/tokenizer/hf_gpt2",
+                truncation=True,
+                padding="max_length",
+                max_length=None,
+                special_tokens={"pad_token": "[PAD]"},
+            ),
+            1024,
+            4,
+        ),
+    ],
+)
+def test_hf_tokenize(
+    text: str,
+    tokenizer_config: PreTrainedHFTokenizerConfig,
+    expected_length: int,
+    expected_num_paddding_tokens: int,
+):
+    # also see here for the truncation and padding options and their effects:
+    # https://huggingface.co/docs/transformers/pad_truncation#padding-and-truncation
+
+    tokenizer_config_dict = tokenizer_config.model_dump()
+    tokenizer = PreTrainedHFTokenizer(**tokenizer_config_dict)
+
+    token_ids = tokenizer.tokenize(text)
+
+    # make sure that the overall token sequence length is correct
+    assert len(token_ids) == expected_length
+
+    # check number of non-padding tokens
+    assert sum(np.array(token_ids) == 43488) == (
+        expected_length - expected_num_paddding_tokens
     )
-    _assert_tokenization(tokenizer)
+
+    # check number of padding tokens
+    assert sum(np.array(token_ids) == 50257) == expected_num_paddding_tokens
```

Comment:
```suggestion
    assert sum(np.array(token_ids) == 50257) == expected_num_padding_tokens
```
Typo, occurrence 3 of 3